src/appleseed.shaders/src/max/as_max_uv_transform.osl

```diff
@@ -0,0 +1,131 @@
+
+//
+// This source file is part of appleseed.
+// Visit http://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2017 Sergo Pogosyan, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+shader as_max_uv_transform
+(
+    float in_coordU = u,
+    float in_coordV = v,
+
+    float in_offsetU = 0.0,
+    float in_offsetV = 0.0,
+
+    float in_tilingU = 1.0,
+    float in_tilingV = 1.0,
+
+    float in_rotateW = 0.0,
+
+    int in_mirrorU = 0,
+    int in_mirrorV = 0,
+    
+    int in_wrapU = 0,
+    int in_wrapV = 0,
+
+    float in_cropW = 1.0,
+    float in_cropH = 1.0,
+    float in_cropU = 0.0,
+    float in_cropV = 0.0,
+
+    string in_crop_mode = "off",
+
+    output float out_U = 0.0,
+    output float out_V = 0.0,
+)
+{
+    float out_u = in_coordU;
+    float out_v = in_coordV;
+    float u_center = 0.5;
+    float v_center = 0.5;
+
+    // Apply rotation before tiling/offset which is different from 3ds max native bitmap transforms
+    if (in_rotateW != 0.0)
+    {
+        point rot = rotate(
```

Comment:
It would make sense to use a specialized 2D rotation function, I'm not sure the OSL compiler would be to optimize this.

Reply:
Is that the one from `appleseed/gaffer/transform.h` header? Looks like it compiles and renders normally. Had to change plugin a little to pass degrees instead of radians.